Add Colab-tutorial

thisiseshan opened the pull request:
link for #64 
I have added the link to Colab tutorial at Hub/examples/mnist/Colab-demo.md

davidbuniat commented:
Hey, @thisiseshan this looks pretty good attempt. As you have noticed the issue was with caching. We have turned off intentionally because it causes issues with some cases (not with training images).

If you add the following line it will make training run faster.
```
from hub.collections.client_manager import HubCache
cache = HubCache(2e9)
cache.register()
```
This issue should be addressed on v1.

Regarding collab feedback. It would be great if you could review other collab examples to make seamless for users to get started and understand the core values of Hub. For reference please take a look at it. I guess adding some explanation would really benefit the user (probably someone who is not aware of what's going on).

- TFDS [https://colab.research.google.com/github/tensorflow/datasets/blob/master/docs/overview.ipynb]
- https://colab.research.google.com/github/huggingface/transformers/blob/master/notebooks/02-transformers.ipynb

davidbuniat commented:
One more remark would be great to have the collab link (with a badge) added to README.md as an example.

thisiseshan commented:
Sure, I will document it well and add the badge. Thank you for the proactive response :)

thisiseshan commented:
I have added the badge in the README.md under the examples (and deleted the Colab-demo.md)
Let me know your thoughts on the updated Colab file. I have added comments to Hub code and basic info about MNIST and working of Colab file.

If there is anything specific I am missing out, do let me know. I will be happy to contribute.

davidbuniat commented:
@thisiseshan This is much better! 

I have had some issues with running the collab `train_set` is not defined.

```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-19-8ec988708080> in <module>()
----> 1 train_dataset = torch.utils.data.Subset(train_set, range(40000))
      2 test_dataset = torch.utils.data.Subset(train_set, range(50000, 60000))
      3 
      4 # Please make sure that collate_fn is provided from the dataset ds.collate_fn to stack tensors together since they are in dictionary form
      5 train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, collate_fn=ds.collate_fn)

NameError: name 'train_set' is not defined
```

thisiseshan commented:
Oh! looks like in my attempt to make the documentation uniform by using `ds`, I missed changing it at `train_dataset` and `test_dataset`.

thisiseshan commented:
I noticed it today, the `dataset.load()` function didn't work without `!pip3 install s3fs`
It is a little strange, as I was able to use it without having the latest s3fs installed.

I have added this code snippet and it works fine. Please let me know if there any changes you wish from my side.

davidbuniat commented:
> I noticed it today, the `dataset.load()` function didn't work without `!pip3 install s3fs`
> It is a little strange, as I was earlier able to use it without having the latest s3fs installed.
> 
> I have added this code snippet and it works fine. Please let me know if there any changes you wish from my side.

Oh, this is strange. Maybe we removed s3fs from requirements or smth recently changed in fsspec. 

Collab looks great! Final remarks.
1) `Turn off caching` -> `Turn on caching`
2) I would move the badge up in readme so it's visible from the start, could go under the logo or maybe under Getting Started.

thisiseshan commented:
> 1. `Turn off caching` -> `Turn on caching`
Done!
> 2. I would move the badge up in readme so it's visible from the start, could go under the logo or maybe under Getting Started.
Done!

thisiseshan commented:
My PR isn't clearing the Locust check. Please guide me through it

davidbuniat commented:
This all looks great! great iterating with you and really thanks @thisiseshan for this awesome job! Would love your help with other issues!

thisiseshan commented:
Thank you @davidbuniat and the community! 🤗

mikayelh commented:
@thisiseshan, thanks for being awesome! we really appreciate the contribution. If you'd like to stay in touch, contribute more or ask us more about anything dataset management/data pipeline-related, please join our community slack [here](https://join.slack.com/share/zt-ihhlezv8-LNcxULTKJMnLrUZQeQdl_w)!
